src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -79,14 +78,10 @@ def match_recommendations(self, message):
         data = json.loads(message.get("data"))
         recommendations = data.get("recommended_matches", [])
         try:
-            validated_records = self._get_validated_bid_offer_match_list(recommendations)
-            for record in validated_records:
-                market = self.markets_mapping.get(record["market_id"])
-                try:
-                    market.match_recommendations([record])
-                except (OfferNotFoundException, BidNotFoundException):
-                    # If the offer or bid have already been matched or deleted.
-                    continue
```

Comment:
Why is this part not needed anymore? Where were the `BidNotFoundException` and `OfferNotFoundException` exceptions raised before?

Reply:
If the bid or offer are not found, we will skip the recommendation in the TwoSided `match_recommendations` function.
However, if at some point the bid/offer were gone after the check, the accept_offer and accept_bid functions will raise `OfferNotFoundException` and `BidNotFoundException` respectively